### PR TITLE
Mobile 4009

### PR DESCRIPTION
**Pull request by crazyserver**

### CODE REVIEW

File: src/core/components/password-modal/password-modal.ts [hunk 1]
```diff
@@ -108,4 +108,5 @@ export type CorePasswordModalResponse = {
     password: string;
     validated?: boolean;
     error?: string;
+    cancel?: boolean;
```
Comment (NoelDeMartin) [inline src/core/components/password-modal/password-modal.ts:111]: I don't think we should declare this property here, given that the password modal is not returning this anywhere. Instead, if you want to extend this type in the course summary usage, you can declare it there like `CorePasswordModalResponse & { cancel?: boolean }` or something.

**Review by NoelDeMartin — CHANGES_REQUESTED** [review]
Acceptance tests were failing, but I'm not sure if the failures were related with these changes so I triggered them again. But if it keeps failing maybe something has introduced a regression (or most likely, the tests need to be updated for the new workflow).

File: src/core/components/password-modal/password-modal.ts (hunk 1 above)
Comment (crazyserver) [reply src/core/components/password-modal/password-modal.ts]: Thanks Done!

**Comment by crazyserver** [comment]
I've added a new function userHasAccessToCourse to let the user enter the course when offline

**Review by NoelDeMartin — APPROVED** [review]

**Comment by phyr0s** [comment]
@NoelDeMartin  @crazyserver  animations still broken in go back action (enter works fine no glitch) due await getting if user is guest  as --> https://github.com/moodlehq/moodleapp/pull/3727

**Comment by NoelDeMartin** [comment]
Thanks for letting us know @phyr0s. We're on low capacity at the moment because many members of the team are on holidays, but we'll take a look after the holiday period.

**Pull request merged**